### ISSUE TITLE
Twitch Adblock

### ISSUE THREAD

**Issue by c133** [pixeltris/TwitchAdSolutions#22]
Hey pixel,

I had been using a really good twitch adblocker that had been working for weeks until today. The extension dissapeared from the both firefox and chrome webstore. Really weird. But I had backed up the chrome extension / exported it as a crx file. 

Are you willing to help me set this up and get it working, I'm willing to pay you. Let me know if there's some way I can contact you besides github. Thanks.

**Comment by ImpalaPUA** [comment]
The extension would still work, you would need the code that chooseneye had on his site choosen.dev

**Comment by c133** [comment]
@ImpalaPUA lol mind read much. Do you know what happened, did he get some kind of cease and desist or something? Have you found a comparable solution? From what you are saying I don't think the crx backup of the chrome extension is enough

**Comment by ImpalaPUA** [comment]
It's not enough no, you should ask him here what happened

https://twitter.com/ChoosenEye

And post it here, since I don't have a twitter

**Comment by ChoosenEye** [comment]
Twitch Adblock is gone.
There will be no new version.
Sorry

**Comment by ImpalaPUA** [comment]
> Twitch Adblock is gone.
> There will be no new version.
> Sorry

Did they lawyer you or was the cost too high to run?

**Comment by c133** [comment]
> Twitch Adblock is gone.
> There will be no new version.
> Sorry

Are you willing to sell your code, your extension literally improved my quality of life I'm not joking

**Comment by ChoosenEye** [comment]
Twitch shut down the way it worked.

Lastly there was 150k users and 6 million views per day

**Comment by ImpalaPUA** [comment]
Didn't it just swap m3u8's?

**Comment by ImpalaPUA** [comment]
You got the m3u8 from lets say a VPN in a country twitch doesn't run ads to, and swapped it when you loaded a twitch stream no?

**Comment by CodeMeZone** [comment]
```

  //fck pixeltris :)
  chrome.cookies.set({
    name: "auth",
    domain: ".choosen.dev",
    httpOnly: true,
    value: Date.now().toString(),
    url: "https://choosen.dev",
  });

  if (match !== null && match.length > 1) {
      var req = new XMLHttpRequest();
      req.open("GET", `https://choosen.dev/stream/auth`, false);

      req.send();

      if (req.status >= 400) {
        return { redirectUrl: details.url };
      } else {
        return { redirectUrl: `https://choosen.dev/stream/${match[1]}` };
      }
    }
}

chrome.webRequest.onBeforeRequest.addListener(
  onBeforeRequest,
  { urls: ["https://usher.ttvnw.net/api/channel/hls/*"] },
  ["blocking", "extraHeaders"]
);
```

Seems like @ChoosenEye dtopped it all.

**Comment by c133** [comment]
I will pay someone to get this working again

**Comment by JigsLove** [comment]
@ChoosenEye Hello Sir, at least tell us the reason why you removed it from the store. It was the best ad blocker for twitch. Thank you Sir.

**Comment by JigsLove** [comment]
@c133 Yes bro, I'm also missing that great extension.

**Comment by finiteLimit** [comment]
I'm curious - what country was the proxy in? From my understanding, this extension is based on fetching the `.m3u8` file using a proxy in another country, a country which does not server Twitch ads. Which country (or countries) would that be?

**Comment by racheljcorey** [comment]
> I'm curious - what country was the proxy in? From my understanding, this extension is based on fetching the `.m3u8` file using a proxy in another country, a country which does not server Twitch ads. Which country (or countries) would that be?

Hopefully this helps someone else out who will continue development on this great extension:

I used my VPN to load Twitch from different countries, and although ads are served to me when connecting to any VPN from within the US, ads are not served when connecting from **Israel**. Hopefully this helps!!

**Comment by guniu** [comment]
Damn, I just realized how unwatchable twitch is.

**Comment by ANaPhOrAX** [comment]
> @ChoosenEye Hello Sir, at least tell us the reason why you removed it from the store. It was the best ad blocker for twitch. Thank you Sir.

why do you think? twitch threatened him, obviously.

**Comment by chrcoluk** [comment]
Why would he start an extension that works and then pull it down without stating a reason?

Clearly someone from twitch/amazon got involved.

**Comment by finiteLimit** [comment]
> 
> 
> > I'm curious - what country was the proxy in? From my understanding, this extension is based on fetching the `.m3u8` file using a proxy in another country, a country which does not server Twitch ads. Which country (or countries) would that be?
> 
> Hopefully this helps someone else out who will continue development on this great extension:
> 
> I used my VPN to load Twitch from different countries, and although ads are served to me when connecting to any VPN from within the US, ads are not served when connecting from **Israel**. Hopefully this helps!!

Do you know if there are any other countries besides Israel which do not receive Twitch ads?

**Comment by pixeltris** [comment]
@finiteLimit people had previously documented geographical locations [here](https://old.reddit.com/r/Twitch/comments/kisdsy/i_did_a_little_test_regarding_ads_on_twitch_and/) and [here](https://old.reddit.com/r/uBlockOrigin/comments/jjesgn/fix_for_ublock_origin_on_twitch_i_updated_the/gj9fumd/). Nothing is fool-proof with regard to VPNing, so you'll need to experiment yourself.

I think it would be useful if there was chrome/firefox extension code in this repo for those who wish to go with m3u8 proxying via VPS/VPN. It should essentially be just [this](https://github.com/pixeltris/TwitchAdSolutions/issues/8#issuecomment-766317343), but needs manifest / changes for firefox.

It's a shame to see `Twitch Adblock` go. Hopefully it come back at some point, or something takes its place.

**Comment by c133** [comment]
@pixeltris Choosen claims the method stopped working which is why he took down the extension. Is there any way to actually verify that this method doesn't work anymore? It's more believable that he got threatened by twitch or amazon judging by the way his extension quickly disappeared from the internet. Anyway I am willing to pay you if you can get that method working again.

**Comment by pixeltris** [comment]
The method used in `Twitch Adblock` wasn't public. It didn't use specific countries to bypass ads and was instead a custom workaround to avoid ads.

I haven't tried VPS/VPN + m3u8 proxy in a while, but it should still work. @ceejayc7 mentioned 3 days ago that it's still working. I'll look at creating some code + simple instructions so people can try themselves with less effort.

I've mentioned previously that having a single server with Turbo access should in theory work to provide m3u8 urls without ads. This could have been the method that `Twitch Adblock` used, I have no idea.

**Comment by WhiteShyGuy** [comment]
ok ok ok

**Comment by yourfriendaaron** [comment]
> > @ChoosenEye Hello Sir, at least tell us the reason why you removed it from the store. It was the best ad blocker for twitch. Thank you Sir.
> 
> why do you think? twitch threatened him, obviously.

or just paid him. no disrespect to the guy if that's the case, though. i'd be pretty tempted, too.

**Comment by AndreyAsadchy** [comment]
So twitch just DMCA'd my android app Xtra for Twitch with 230k users after i published a beta update testing my new adblock method. I guess the reason is because i was dumb enough to mention it in the changelog, since other alternative apps are fine.

**Comment by racheljcorey** [comment]
> 
> 
> So twitch just DMCA'd my android app Xtra for Twitch with 230k users after i published a beta update testing my new adblock method. I guess the reason is because i was dumb enough to mention it in the changelog, since other alternative apps are fine.

That SUCKS!!! I loved using Xtra too!!! I'm so sorry to hear that. :(

**Comment by llacb47** [comment]
@AndreyAsadchy dang. Did they specifically mention that in the notice?

**Comment by AndreyAsadchy** [comment]
@llacb47 No, only that the app "infringes Twitch’s copyrights to exclusive broadcast content from www.twitch.tv as well as copyrights in Twitch’s Developer Materials made available under the Twitch Developer Services Agreement".

**Comment by bingoxo** [comment]
don't know if it's useful for anyone but here is the latest version before he removed it from chrome store 

https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w

**Comment by CodeMeZone** [comment]
Does anyone know how to make something like this?

I wouldn't mind throwing a few server up

**Comment by JigsLove** [comment]
@bingoxo Thank you brother for sharing. <3

**Comment by Manderville** [comment]
What do I use to block Twitch ads now?

**Comment by deaglenomics** [comment]
Twitch prime ...  j/k

Some people have had some success with alternate player for twitch.

**Comment by hypeincarnate** [comment]
> So twitch just DMCA'd my android app Xtra for Twitch with 230k users after i published a beta update testing my new adblock method. I guess the reason is because i was dumb enough to mention it in the changelog, since other alternative apps are fine.

Do you have the update for Xtra?  I've been trying to find a way to watch twitch mobile without 5 mins of ads.

**Comment by PussyCatt** [comment]
As you know, if not continue to sleep and let the “big guys” do it, the rich (stars, politicians, doctors, lawyers, judges, “business” men / women) control this planet. Countries, territories, states, cities, etc. do not really exist. What exists are the multinationals, the rich, who decide everything.

And when a little man wants to go against their "plans" (to steal the population - the capitalist is the only system that exists now and it is perfect for the rich) they organize themselves so that this person cannot " to bother ".

In the case that interests us here, no doubt the programmer was bought by Amazon (which owns Twitch and many other sub-divisions).

When it is more "serious", the rich do not hesitate to commit murders as they do in the field of health, by making sure to screw everything to the "private", thanks to their puppets the politicians, small rich.
Then the rich have free access to all unimaginable health services because it is the population who pays for them. And that same population, even though it pays for all "private" services, is on endless waiting lists and dies before seeing a doctor.

But the population does not care because it takes 55 years and over to worry about health.

In short, the Capitalist is the best of the world… for the rich.
And the people continue to revere them and will worship them as long as humans exist because the intelligence of humans is on the same level as a chair.

**Comment by TheOldOne822** [comment]
> don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> 
> https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w

The file has been removed from Mega upload.

**Comment by jsgueglia84** [comment]
> > don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> > https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w
> 
> The file has been removed from Mega upload.

post new ? does old version still work?

**Comment by arthurbolsoni** [comment]
if buy the twitch turbo can sign the urls and return by a server. this could work i think

**Comment by jsgueglia84** [comment]
> People posting Chrome version but nothing about Firefox version... Well, here, in this forgotten browser, everything works good.
> https://drive.google.com/file/d/1TyREnxhmEbPDuyonrByHiOh7gYS133Gk/view?usp=sharing

The chrome version posted is a dead URL!!!! Can you please post Chrome?

**Comment by gatsuslayer** [comment]
> don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> 
> https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w

Can you reupload it again? the link is down

**Comment by bingoxo** [comment]
> > don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> > https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w
> 
> The file has been removed from Mega upload.



> > > don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> > > https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w
> > 
> > 
> > The file has been removed from Mega upload.
> 
> post new ? does old version still work?



> > People posting Chrome version but nothing about Firefox version... Well, here, in this forgotten browser, everything works good.
> > https://drive.google.com/file/d/1TyREnxhmEbPDuyonrByHiOh7gYS133Gk/view?usp=sharing
> 
> The chrome version posted is a dead URL!!!! Can you please post Chrome?



> > People posting Chrome version but nothing about Firefox version... Well, here, in this forgotten browser, everything works good.
> > https://drive.google.com/file/d/1TyREnxhmEbPDuyonrByHiOh7gYS133Gk/view?usp=sharing
> 
> The chrome version posted is a dead URL!!!! Can you please post Chrome?



> > don't know if it's useful for anyone but here is the latest version before he removed it from chrome store
> > https://mega.nz/file/fU9wzCBI#YZOjHh2qw-CjMFuo-ahErAvuwLxxi9_s6b3yOedoR7w
> 
> Can you reupload it again? the link is down

Link updated

**Comment by pixeltris** [comment]
It's probably best to not use external links as they can die (as already observed). I'll reupload both of the above (encased in a zip as required by github):

chrome taken from the post by @bingoxo
[extension_2_2_0_0.zip](https://github.com/pixeltris/TwitchAdSolutions/files/6278901/extension_2_2_0_0.zip)

firefox taken from the post by @Kaederoth
[{dfabfeb8-4edf-4296-ba99-de1bc071c7c3}.zip](https://github.com/pixeltris/TwitchAdSolutions/files/6278905/dfabfeb8-4edf-4296-ba99-de1bc071c7c3.zip)

**Comment by c133** [comment]
> It's probably best to not use external links as they can die (as already observed). I'll reupload both of the above (encased in a zip as required by github):
> 
> chrome taken from the post by @bingoxo
> [extension_2_2_0_0.zip](https://github.com/pixeltris/TwitchAdSolutions/files/6278901/extension_2_2_0_0.zip)
> 
> firefox taken from the post by @Kaederoth
> [{dfabfeb8-4edf-4296-ba99-de1bc071c7c3}.zip](https://github.com/pixeltris/TwitchAdSolutions/files/6278905/dfabfeb8-4edf-4296-ba99-de1bc071c7c3.zip)

Aren't these extensions essentially useless since we don't have Choosen's server side code?

**Comment by pixeltris** [comment]
@c133 yes, however it's a useful starting point for those who want to host their own m3u8 proxy server.

**Comment by arthurbolsoni** [comment]
i am trying replicate the server. but i had receiving the CORS error on set in twitch.

**Comment by CodeMeZone** [comment]
> i am trying replicate the server. but i had receiving the CORS error on set in twitch.

Maybe we can get a github going so some of us can help out?

**Comment by arthurbolsoni** [comment]
call me on discord @knightbrowser#9246

**Comment by FrostySo** [comment]
> So twitch just DMCA'd my android app Xtra for Twitch with 230k users after i published a beta update testing my new adblock method. I guess the reason is because i was dumb enough to mention it in the changelog, since other alternative apps are fine.

Hey man, I hope you can find a solution to bring this app back. I use this pretty much daily. One of the best twitch apps hands down. Perhaps you could have some sort of plugin support, that gives you access to hmm I don't know, intercept the network in order to create an adblock plugin. That might in theory, get rid of future DMCA requests from twitch. I'm no lawyer or anything, so I don't know if it's possible for you to bring this app back under the same name, or if at all (I hope you can). 

I cannot believe twitch is going through all this trouble to spam everyone with ads. They are really irritating and I no longer use twitch because of it (unless it is resolved again). 

Sucks that this world has come to this. Twitch should focus on making their platform better, instead of worse.

**Comment by arthurbolsoni** [comment]
someone now a proxy/vps for i purchase in country without ads?

**Comment by bingoxo** [comment]
> someone know a proxy/vps for i purchase in country without ads?

Lowendbox.com

**Comment by revunix** [comment]
I have two very cheap KVM servers in Russia and Ukraine. 

- https://hostvds.com/
- https://ruvps.net/

**Comment by jamesdrudolph** [comment]
> 
> 
> someone know a proxy/vps for i purchase in country without ads?

Amazon AWS free-tier in Bahrain does not give ads, I have been using it for 1-2 weeks now and have not any.

**Comment by AndreyAsadchy** [comment]
> > So twitch just DMCA'd my android app Xtra for Twitch with 230k users after i published a beta update testing my new adblock method. I guess the reason is because i was dumb enough to mention it in the changelog, since other alternative apps are fine.
> 
> Hey man, I hope you can find a solution to bring this app back. I use this pretty much daily. One of the best twitch apps hands down. Perhaps you could have some sort of plugin support, that gives you access to hmm I don't know, intercept the network in order to create an adblock plugin. That might in theory, get rid of future DMCA requests from twitch. I'm no lawyer or anything, so I don't know if it's possible for you to bring this app back under the same name, or if at all (I hope you can).
> 
> I cannot believe twitch is going through all this trouble to spam everyone with ads. They are really irritating and I no longer use twitch because of it (unless it is resolved again).
> 
> Sucks that this world has come to this. Twitch should focus on making their platform better, instead of worse.

Google denied my appeal. If you want, you can follow my subreddit r/XtraForTwitch, maybe i will upload the app to other stores like from samsung etc.

**Comment by pixeltris** [comment]
https://github.com/TTV-LOL/extensions this popped up which looks like a direct replacement for `Twitch Adblock`. They mention [here](https://old.reddit.com/r/Adblock/comments/modkbv/we_made_a_new_twitch_ad_block_extension_to/) that Firefox is coming soon.

I've listed it as the recommended solution in the README due to the method used, but I haven't personally tried it myself.

@AndreyAsadchy if you could ping me when there are links available, that would be useful to add to the `Other solutions` in the README of this repo. Not sure why I didn't have it there whilst it was still on Google Play...

**Comment by c133** [comment]
> https://github.com/TTV-LOL/extensions this popped up which looks like a direct replacement for `Twitch Adblock`. They mention [here](https://old.reddit.com/r/Adblock/comments/modkbv/we_made_a_new_twitch_ad_block_extension_to/) that Firefox is coming soon.
> 
> I've listed it as the recommended solution in the README due to the method used, but I haven't personally tried it myself.
> 
> @AndreyAsadchy if you could ping me when there are links available, that would be useful to add to the `Other solutions` in the README of this repo. Not sure why I didn't have it there whilst it was still on Google Play...

Thank you <3 trying it now

**Comment by FrostySo** [comment]
> Google denied my appeal. If you want, you can follow my subreddit r/XtraForTwitch, maybe i will upload the app to other stores like from samsung etc.

Sure. Followed. 

Hopefully you give a direct download as well, cause I don't own a samsung phone and don't plan on owning one. If you want, maybe possibly open source your project (or partially have it open) for others to help contribute with it. It really has a lot of potential! I really hate twitch :(

**Comment by AlexWatsonnn** [comment]
Have you tried any of these solutions:

- Go to another browser
- Update the AdBlock extension
- Use the emulator to run Twitch on PC
- Make the required changes in the Chrome browser
- Try another extension

For full detail, visit this page: **[AdBlock Stopped Working on Twitch](https://www.pcerror-fix.com/fix-adblock-not-working-on-twitch)**

**Comment by arthurbolsoni** [comment]
> Have you tried any of these solutions:
> 
> * Go to another browser
> * Update the AdBlock extension
> * Use the emulator to run Twitch on PC
> * Make the required changes in the Chrome browser
> * Try another extension
> 
> For full detail, visit this page: **[AdBlock Stopped Working on Twitch](https://www.pcerror-fix.com/fix-adblock-not-working-on-twitch)**

it still show the purple screen no?

**Comment by arthurbolsoni** [comment]
https://github.com/arthurbolsoni/Purple-adblock/

i do that extension and i am doing for keep online. i gonna publish the code too. it is in firefox and google store. pls help me with code or donation

**Comment by revunix** [comment]
@arthurbolsoni very nice! which server location and which provider?

**Comment by arthurbolsoni** [comment]
> @arthurbolsoni very nice! which server location and which provider?

i paying a from russia. but i am open for another ideia location.

**Comment by revunix** [comment]
> i paying a from russia. but i am open for another ideia location.

Thanks fyi :)

**Comment by arthurbolsoni** [comment]
> Thanks fyi :)

If you find bug pls report. i am with lot of ideias for the project

**Comment by pixeltris** [comment]
Closing this. As mentioned there are now two direct replacements for `Twitch Adblock` (`TTV.LOL` / `Purple AdBlock`) which are linked in the README.
